narwhals/functions.py

```diff
@@ -904,6 +904,18 @@ def _scan_csv_impl(
         from pyarrow import csv  # ignore-banned-import
 
         native_frame = csv.read_csv(source, **kwargs)
+    elif implementation.is_spark_like():
+        if (session := kwargs.pop("session", None)) is None:
+            msg = "Spark like backends require a session object to be passed in `kwargs`."
+            raise ValueError(msg)
+        native_frame = (
+            session.read.format("csv").load(source)
+            if (
+                implementation is Implementation.SQLFRAME
+                and (parse_version(version("sqlframe"))) < (3, 27, 0)
+            )
+            else session.read.format("csv").options(**kwargs).load(source)
+        )
```

Comment:
```suggestion
        reader = session.read.format("csv")
        if not (
            implementation is Implementation.SQLFRAME
            and (parse_version(version("sqlframe"))) < (3, 27, 0)
        ):
            reader = reader.options(**kwargs)
        native_frame = reader.load(source)
```

I remember you commenting about readibility in the previous `scan_parquet` PR. I just thought about this alternative (but if we change here we should probably make `scan_parquet` consistent)

Reply:
You could also use the method `is_sqlframe`

```py
implementation is Implementation.SQLFRAME
implementation.is_sqlframe()
```